test/libsolidity/semanticTests/constructor/constructor_function_complex.sol

```diff
@@ -0,0 +1,22 @@
+contract D {
+    uint public x;
+    constructor(function() external pure returns (uint) g) {
+        x = g();
+    }
+}
+
+contract C {
+    function f() public returns (uint r) {
+        D d = new D(this.sixteen);
+        r = d.x();
+    }
+
+    function sixteen() public pure returns (uint) {
+        return 16;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor() ->
```

Comment:
Why is the constructor call needed here? Shouldn't the call to f() below construct D?

Reply:
The constructor call is not needed here. I'll just remove it then.